events/sns.go

```diff
@@ -18,15 +18,20 @@ type SNSEventRecord struct {
 }
 
 type SNSEntity struct {
-	Signature         string                 `json:"Signature"`
-	MessageID         string                 `json:"MessageId"`
-	Type              string                 `json:"Type"`
-	TopicArn          string                 `json:"TopicArn"`
-	MessageAttributes map[string]interface{} `json:"MessageAttributes"`
-	SignatureVersion  string                 `json:"SignatureVersion"`
-	Timestamp         time.Time              `json:"Timestamp"`
-	SigningCertURL    string                 `json:"SigningCertUrl"`
-	Message           string                 `json:"Message"`
-	UnsubscribeURL    string                 `json:"UnsubscribeUrl"`
-	Subject           string                 `json:"Subject"`
+	Signature         string                         `json:"Signature"`
+	MessageID         string                         `json:"MessageId"`
+	Type              string                         `json:"Type"`
+	TopicArn          string                         `json:"TopicArn"`
+	MessageAttributes map[string]SNSMessageAttribute `json:"MessageAttributes"`
+	SignatureVersion  string                         `json:"SignatureVersion"`
+	Timestamp         time.Time                      `json:"Timestamp"`
+	SigningCertURL    string                         `json:"SigningCertUrl"`
+	Message           string                         `json:"Message"`
+	UnsubscribeURL    string                         `json:"UnsubscribeUrl"`
+	Subject           string                         `json:"Subject"`
+}
+
+type SNSMessageAttribute struct {
```

Comment:
Type is missing `Name` per the docs: https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html

Reply:
oops, misunderstood, the key is the name :)